tools/lint_diff.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+import os
+import sys
+import subprocess
+
+CONFIG = os.path.join(
+    os.path.abspath(os.path.dirname(__file__)),
+    'lint_diff.ini',
+)
+
+
+def rev_list(branch, num_commits):
+    """List commits in reverse chronological order.
+
+    Only the first `num_commits` are shown.
+
+    """
+    res = subprocess.run(
+        [
+            'git',
+            'rev-list',
+            '--max-count',
+            f'{num_commits}',
+            '--first-parent',
+            branch
+        ],
+        stdout=subprocess.PIPE,
+        encoding='utf-8',
+    )
+    res.check_returncode()
+    return res.stdout.rstrip('\n').split('\n')
+
+
+def find_branch_point():
+    """Find when the current branch split off master.
+
+    It is based off of this Stackoverflow post:
+
+    https://stackoverflow.com/questions/1527234/finding-a-branch-point-with-git#4991675
+
+    """
+    branch_commits = rev_list('HEAD', 1000)
+    master_commits = set(rev_list('master', 1000))
+    for branch_commit in branch_commits:
+        if branch_commit in master_commits:
+            return branch_commit
+
+    # If a branch split off over 1000 commits ago we will fail to find
+    # the ancestor.
+    raise RuntimeError(
+        'Failed to find a common ancestor in the last 1000 commits')
+
+
+def find_diff(sha):
+    """Find the diff since the given sha."""
+    res = subprocess.run(
+        ['git', 'diff', '--unified=0', sha, '--', '*.py'],
+        stdout=subprocess.PIPE,
+        encoding='utf-8',
+    )
+    res.check_returncode()
+    return res.stdout
+
+
+def run_pycodestyle(diff):
+    """Run pycodestyle on the given diff."""
+    res = subprocess.run(
+        ['pycodestyle', '--diff', '--config', CONFIG],
+        input=diff,
+        stdout=subprocess.PIPE,
+        encoding='utf-8',
+    )
+    return res.returncode, res.stdout
+
+
+def main():
+    branch_point = find_branch_point()
+    diff = find_diff(branch_point)
+    rc, errors = run_pycodestyle(diff)
+    if errors:
+        print(errors)
+    sys.exit(rc)
```

Comment:
if no issues present still might be nice to see a message, something like "No pycodestyle issues found" or something like that, this need not block the PR to go in though.

Reply:
This makes sense when running manually (but then you already know you're running it anyway); I think the same can be said for the `if errors` block though, some header or footer could help. Anyway, this is polish we can do if it's confusing. Let's merge this as is and see if it works without surprises first:)

In CI this gets printed:
```
$ tools/lint_diff.py
The command "tools/lint_diff.py" exited with 0.
```